[Ingest Pipelines] Processor forms for processors E-J

jloleysens opened the pull request:
## Summary

This contribution adds forms for the following processors:

* Enrich
* Fail
* Foreach
* GeoIP
* Grok
* Gsub
* HTML strip
* Inference
* Join
* JSON

## How to test

1. Start Kibana with basic license
2. Go to ingest pipelines plugin in management section
3. Add a processor for each of the forms listed above (attempt to submit to see which fields are required)
4. Create the pipeline

## Additional Context

Continuation of https://github.com/elastic/kibana/pull/72849

### Checklist

Delete any items that are not applicable to this PR.

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68176/)
* Commit: ac6483eb47c1cef29efaffa14cc1965af1dc7d8a
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68176/execution/node/394/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68176/","number":"68176","commit":"ac6483eb47c1cef29efaffa14cc1965af1dc7d8a"}]}
PIPELINE-->

jloleysens commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68359/)
* Commit: 469d8040b37395d3c1fdbe0d68b3efb4652b9a3b

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/469d8040b37395d3c1fdbe0d68b3efb4652b9a3b) | diff | [baseline](https://github.com/elastic/kibana/commit/a9860f3675ac9d46da42d00795704264830a436b) |
| --- | --- | --- | --- |
| ingestPipelines | 401 | +11 | 390 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/469d8040b37395d3c1fdbe0d68b3efb4652b9a3b) | diff | [baseline](https://github.com/elastic/kibana/commit/a9860f3675ac9d46da42d00795704264830a436b) |
| --- | --- | --- | --- |
| ingestPipelines | 674.4KB | +30.5KB | 643.9KB |

### History
* :broken_heart: [Build #68176](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68176/) failed ac6483eb47c1cef29efaffa14cc1965af1dc7d8a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68176/","number":"68176","commit":"ac6483eb47c1cef29efaffa14cc1965af1dc7d8a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68359/","number":"68359","commit":"469d8040b37395d3c1fdbe0d68b3efb4652b9a3b"}]}
PIPELINE-->

jloleysens commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68703/)
* Commit: 4592909a99f821039f6a6a5dc342511d056e32ae

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/4592909a99f821039f6a6a5dc342511d056e32ae) | diff | [baseline](https://github.com/elastic/kibana/commit/e077a2892813652aea040ba0ee5d4dfb736ab88b) |
| --- | --- | --- | --- |
| ingestPipelines | 401 | +11 | 390 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/4592909a99f821039f6a6a5dc342511d056e32ae) | diff | [baseline](https://github.com/elastic/kibana/commit/e077a2892813652aea040ba0ee5d4dfb736ab88b) |
| --- | --- | --- | --- |
| ingestPipelines | 674.0KB | +30.5KB | 643.5KB |

### History
* :green_heart: [Build #68359](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68359/) succeeded 469d8040b37395d3c1fdbe0d68b3efb4652b9a3b
* :broken_heart: [Build #68176](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68176/) failed ac6483eb47c1cef29efaffa14cc1965af1dc7d8a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68176/","number":"68176","commit":"ac6483eb47c1cef29efaffa14cc1965af1dc7d8a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68359/","number":"68359","commit":"469d8040b37395d3c1fdbe0d68b3efb4652b9a3b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68703/","number":"68703","commit":"4592909a99f821039f6a6a5dc342511d056e32ae"}]}
PIPELINE-->

jloleysens commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68982/)
* Commit: 3e83391bf2ca98da7278eadf0af8967415e39518

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/3e83391bf2ca98da7278eadf0af8967415e39518) | diff | [baseline](https://github.com/elastic/kibana/commit/de4b502cb31855b2c9c062ae97f99228138af27e) |
| --- | --- | --- | --- |
| ingestPipelines | 401 | +11 | 390 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/3e83391bf2ca98da7278eadf0af8967415e39518) | diff | [baseline](https://github.com/elastic/kibana/commit/de4b502cb31855b2c9c062ae97f99228138af27e) |
| --- | --- | --- | --- |
| ingestPipelines | 673.9KB | +30.5KB | 643.4KB |

### History
* :green_heart: [Build #68703](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68703/) succeeded 4592909a99f821039f6a6a5dc342511d056e32ae
* :green_heart: [Build #68359](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68359/) succeeded 469d8040b37395d3c1fdbe0d68b3efb4652b9a3b
* :broken_heart: [Build #68176](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68176/) failed ac6483eb47c1cef29efaffa14cc1965af1dc7d8a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68176/","number":"68176","commit":"ac6483eb47c1cef29efaffa14cc1965af1dc7d8a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68359/","number":"68359","commit":"469d8040b37395d3c1fdbe0d68b3efb4652b9a3b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68703/","number":"68703","commit":"4592909a99f821039f6a6a5dc342511d056e32ae"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68982/","number":"68982","commit":"3e83391bf2ca98da7278eadf0af8967415e39518"}]}
PIPELINE-->

jloleysens commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/inference.tsx:
```diff
@@ -0,0 +1,201 @@
+      defaultMessage="Contains the inference type and its options. There are two types: {regression} and {classification}."
+      values={{
+        regression: (
+          <EuiLink
```
nice! did not know about this one.

jloleysens commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/inference.tsx:
```diff
@@ -0,0 +1,201 @@
+        helpText={
+          <FormattedMessage
+            id="xpack.ingestPipelines.pipelineEditor.inferenceForm.targetFieldHelpText"
+            defaultMessage="Field added to incoming documents to contain results objects. Default value is {targetField}."
```
Yeah, some of these are a bit sketchy. I took most of these directly from the docs site, only lightly massaging to fit into context occasionally. These will definitely benefit from copy review.

jloleysens commented:
Thanks for the review @alisonelizabeth ! I addressed your feedback. Do you think you could take another look? I was not able to reproduce the foreach editor bug, but I think it may have been fixed in this PR https://github.com/elastic/kibana/pull/75160.

Would you mind testing again now that I merged master? If the issue resurfaces I will take another look!

alisonelizabeth commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/enrich.tsx:
```diff
@@ -0,0 +1,226 @@
+      'xpack.ingestPipelines.pipelineEditor.enrichForm.maxMatchesFieldHelpText',
+      {
+        defaultMessage:
+          'The maximum number of matched documents to include under the configured target field. The target_field will be turned into a json array if max_matches is higher than 1, otherwise target_field will become a json object. Accepts numbers are 1 up to 128.',
```
"Accepted numbers are 1 up to 128" or "Accepts numbers 1 up to 128"?

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69108/)
* Commit: 6ecf83090df2283b93cafa142fb275a80a9d5f49

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/6ecf83090df2283b93cafa142fb275a80a9d5f49) | diff | [baseline](https://github.com/elastic/kibana/commit/de4b502cb31855b2c9c062ae97f99228138af27e) |
| --- | --- | --- | --- |
| ingestPipelines | 401 | +11 | 390 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/6ecf83090df2283b93cafa142fb275a80a9d5f49) | diff | [baseline](https://github.com/elastic/kibana/commit/de4b502cb31855b2c9c062ae97f99228138af27e) |
| --- | --- | --- | --- |
| ingestPipelines | 673.9KB | +30.5KB | 643.4KB |

### History
* :green_heart: [Build #68982](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68982/) succeeded 3e83391bf2ca98da7278eadf0af8967415e39518
* :green_heart: [Build #68703](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68703/) succeeded 4592909a99f821039f6a6a5dc342511d056e32ae
* :green_heart: [Build #68359](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68359/) succeeded 469d8040b37395d3c1fdbe0d68b3efb4652b9a3b
* :broken_heart: [Build #68176](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68176/) failed ac6483eb47c1cef29efaffa14cc1965af1dc7d8a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68176/","number":"68176","commit":"ac6483eb47c1cef29efaffa14cc1965af1dc7d8a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68359/","number":"68359","commit":"469d8040b37395d3c1fdbe0d68b3efb4652b9a3b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68703/","number":"68703","commit":"4592909a99f821039f6a6a5dc342511d056e32ae"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68982/","number":"68982","commit":"3e83391bf2ca98da7278eadf0af8967415e39518"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69108/","number":"69108","commit":"6ecf83090df2283b93cafa142fb275a80a9d5f49"}]}
PIPELINE-->

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/gsub.tsx:
```diff
@@ -7,88 +7,72 @@
+      {
+        validator: emptyField(
+          i18n.translate('xpack.ingestPipelines.pipelineEditor.gsubForm.patternRequiredError', {
+            defaultMessage: 'A pattern value is required.',
```
```suggestion
            defaultMessage: 'A value is required.',
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/gsub.tsx:
```diff
@@ -7,88 +7,72 @@
+      {
+        validator: emptyField(
+          i18n.translate('xpack.ingestPipelines.pipelineEditor.gsubForm.replacementRequiredError', {
+            defaultMessage: 'A replacement value is required.',
```
```suggestion
            defaultMessage: 'A value is required.',
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/enrich.tsx:
```diff
@@ -0,0 +1,226 @@
+      return n === 1 ? undefined : n;
+    },
+    label: i18n.translate('xpack.ingestPipelines.pipelineEditor.enrichForm.maxMatchesFieldLabel', {
+      defaultMessage: 'Max matches (optional)',
```
```suggestion
      defaultMessage: 'Maximum matches (optional)',
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/html_strip.tsx:
```diff
@@ -0,0 +1,37 @@
+      <FieldNameField
+        helpText={i18n.translate(
+          'xpack.ingestPipelines.pipelineEditor.htmlStripForm.fieldNameHelpText',
+          { defaultMessage: 'The string-valued field to remove HTML tags from.' }
```
```suggestion
          { defaultMessage: 'Field to remove HTML tags from' }
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/html_strip.tsx:
```diff
@@ -0,0 +1,37 @@
+          'xpack.ingestPipelines.pipelineEditor.htmlStripForm.targetFieldHelpText',
+          {
+            defaultMessage:
+              'The field to assign the stripped value to. If blank, the field will be updated in-place.',
```
Can you replace the `{{original field}}` reference here?
```suggestion
              'Field used to contain stripped text. Defaults to {{original field}}.',
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/inference.tsx:
```diff
@@ -0,0 +1,203 @@
+    helpText: i18n.translate(
+      'xpack.ingestPipelines.pipelineEditor.inferenceForm.modelIDFieldHelpText',
+      {
+        defaultMessage: 'The ID of the model to load and infer against.',
```
```suggestion
        defaultMessage: 'ID of the model to infer against',
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/inference.tsx:
```diff
@@ -0,0 +1,203 @@
+      'xpack.ingestPipelines.pipelineEditor.inferenceForm.fieldMapHelpText',
+      {
+        defaultMessage:
+          'Maps the document field names to the known field names of the model. This mapping takes precedence over any default mappings provided in the model configuration.',
```
```suggestion
          'Maps document field names to the known field names of the model. Takes precedence over any mappings in the model.',
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/inference.tsx:
```diff
@@ -0,0 +1,203 @@
+        helpText={
+          <FormattedMessage
+            id="xpack.ingestPipelines.pipelineEditor.inferenceForm.targetFieldHelpText"
+            defaultMessage="Field added to incoming documents that will contain inference processor results. Default value is {targetField}."
```
```suggestion
            defaultMessage="Field used to contain inference processor results. Defaults to {targetField}."
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/join.tsx:
```diff
@@ -0,0 +1,67 @@
+    helpText: i18n.translate(
+      'xpack.ingestPipelines.pipelineEditor.joinForm.separatorFieldHelpText',
+      {
+        defaultMessage: 'The separator character.',
```
```suggestion
        defaultMessage: 'Separator character',
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/join.tsx:
```diff
@@ -0,0 +1,67 @@
+      <FieldNameField
+        helpText={i18n.translate(
+          'xpack.ingestPipelines.pipelineEditor.joinForm.fieldNameHelpText',
+          { defaultMessage: 'The field to be separated.' }
```
```suggestion
          { defaultMessage: 'Field containing array values to join' }
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/join.tsx:
```diff
@@ -0,0 +1,67 @@
+          'xpack.ingestPipelines.pipelineEditor.joinForm.targetFieldHelpText',
+          {
+            defaultMessage:
+              'The field to assign the joined value to. If empty, the field is updated in-place.',
```
```suggestion
              'Field used to contain the joined value. Defaults to {{original field}}.',
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/json.tsx:
```diff
@@ -0,0 +1,82 @@
+      'xpack.ingestPipelines.pipelineEditor.jsonForm.addToRootFieldHelpText',
+      {
+        defaultMessage:
+          'Inject the serialized JSON to the top level of the JSON document. A target field cannot be combined with this option.',
```
```suggestion
          'If true, add the JSON object to the top level of the document. Cannot be combined with a target field.',
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/json.tsx:
```diff
@@ -0,0 +1,82 @@
+
+  return (
+    <>
+      <FieldNameField
+        helpText={i18n.translate(
+          'xpack.ingestPipelines.pipelineEditor.joinForm.fieldNameHelpText',
+          { defaultMessage: 'The field to be separated.' }
+        )}
+      />
+
+      <TargetField />
```
This looks like a copy/paste error.

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/geoip.tsx:
```diff
@@ -0,0 +1,101 @@
+    <>
+      <FieldNameField
+        helpText={i18n.translate(
+          'xpack.ingestPipelines.pipelineEditor.grokForm.fieldNameHelpText',
```
This ref looks wrong.

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/shared/map_processor_type_to_form.tsx:
```diff
@@ -106,63 +116,70 @@ export const mapProcessorTypeToDescriptor: MapProcessorTypeToDescriptor = {
   enrich: {
-    FieldsComponent: undefined, // TODO: Implement
+    FieldsComponent: Enrich,
     docLinkPath: '/enrich-processor.html',
```
Forgot to add this in my original review:

Unlike the other processors, the enrich processor is a bit complicated. I'd link to these docs rather than the processor docs:
```suggestion
    docLinkPath: '/ingest-enriching-data.htmll',
```

jloleysens commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/enrich.tsx:
```diff
@@ -0,0 +1,226 @@
+      defaultMessage: 'Policy name',
+    }),
+    helpText: i18n.translate('xpack.ingestPipelines.pipelineEditor.enrichForm.policyNameHelpText', {
+      defaultMessage: 'The name of the enrich policy to use.',
```
Great point, will do!

jloleysens replied:
Great point, will do!

jloleysens commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/enrich.tsx:
```diff
@@ -0,0 +1,226 @@
+      'xpack.ingestPipelines.pipelineEditor.enrichForm.shapeRelationFieldHelpText',
+      {
+        defaultMessage:
+          'A spatial relation operator used to match the geo_shape of incoming documents to documents in the enrich index. This option is only used for geo_match enrich policy types.',
```
Will do!

jloleysens commented:
@elasticmachine merge upstream

jloleysens commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/grok.tsx:
```diff
@@ -0,0 +1,159 @@
+    ),
+  },
+
+  properties: {
+    type: FIELD_TYPES.COMBO_BOX,
+    deserializer: to.arrayOfStrings,
+    label: i18n.translate('xpack.ingestPipelines.pipelineEditor.grokForm.propertiesFieldLabel', {
+      defaultMessage: 'Properties (optional)',
+    }),
+    helpText: i18n.translate(
+      'xpack.ingestPipelines.pipelineEditor.grokForm.propertiesFieldHelpText',
+      {
+        defaultMessage:
+          'Controls what properties are added to the target field. Values depend on what is available from the database file.',
+      }
+    ),
+  },
+
+  first_only: {
+    type: FIELD_TYPES.TOGGLE,
+    defaultValue: true,
+    deserializer: to.booleanOrUndef,
+    serializer: from.defaultBoolToUndef(true),
+    label: i18n.translate('xpack.ingestPipelines.pipelineEditor.grokForm.firstOnlyFieldLabel', {
+      defaultMessage: 'First only',
+    }),
+  },
+};
+
+export const Grok: FunctionComponent = () => {
+  return (
+    <>
+      <FieldNameField
+        helpText={i18n.translate(
+          'xpack.ingestPipelines.pipelineEditor.grokForm.fieldNameHelpText',
+          { defaultMessage: 'The array field.' }
+        )}
+      />
+
```
Thanks for spotting this! They are indeed, these fields were not being rendered in the UI, just copy/paste that was not cleaned up for field configuration.

jloleysens commented:
@jrodewig Thanks for the review!! I've addressed your feedback, would you mind taking another look?

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69322/)
* Commit: 1ad0f2e7139c9214197531c17f229b9a75fe117b
* This comment will update when the build is complete

### Failed CI Steps
* [Build X-Pack Kibana](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69322/execution/node/308/log)

### History
* :green_heart: [Build #69108](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69108/) succeeded 6ecf83090df2283b93cafa142fb275a80a9d5f49
* :green_heart: [Build #68982](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68982/) succeeded 3e83391bf2ca98da7278eadf0af8967415e39518
* :green_heart: [Build #68703](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68703/) succeeded 4592909a99f821039f6a6a5dc342511d056e32ae
* :green_heart: [Build #68359](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68359/) succeeded 469d8040b37395d3c1fdbe0d68b3efb4652b9a3b
* :broken_heart: [Build #68176](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68176/) failed ac6483eb47c1cef29efaffa14cc1965af1dc7d8a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68176/","number":"68176","commit":"ac6483eb47c1cef29efaffa14cc1965af1dc7d8a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68359/","number":"68359","commit":"469d8040b37395d3c1fdbe0d68b3efb4652b9a3b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68703/","number":"68703","commit":"4592909a99f821039f6a6a5dc342511d056e32ae"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68982/","number":"68982","commit":"3e83391bf2ca98da7278eadf0af8967415e39518"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69108/","number":"69108","commit":"6ecf83090df2283b93cafa142fb275a80a9d5f49"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69322/","number":"69322","commit":"1ad0f2e7139c9214197531c17f229b9a75fe117b"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69325/)
* Commit: 4030a9966211ae6344037657b8f12a78146504c3
* This comment will update when the build is complete

### Failed CI Steps
* [Build X-Pack Kibana](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69325/execution/node/308/log)

### History
* :broken_heart: [Build #69322](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69322/) failed 1ad0f2e7139c9214197531c17f229b9a75fe117b
* :green_heart: [Build #69108](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69108/) succeeded 6ecf83090df2283b93cafa142fb275a80a9d5f49
* :green_heart: [Build #68982](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68982/) succeeded 3e83391bf2ca98da7278eadf0af8967415e39518
* :green_heart: [Build #68703](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68703/) succeeded 4592909a99f821039f6a6a5dc342511d056e32ae
* :green_heart: [Build #68359](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68359/) succeeded 469d8040b37395d3c1fdbe0d68b3efb4652b9a3b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68359/","number":"68359","commit":"469d8040b37395d3c1fdbe0d68b3efb4652b9a3b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68703/","number":"68703","commit":"4592909a99f821039f6a6a5dc342511d056e32ae"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68982/","number":"68982","commit":"3e83391bf2ca98da7278eadf0af8967415e39518"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69108/","number":"69108","commit":"6ecf83090df2283b93cafa142fb275a80a9d5f49"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69322/","number":"69322","commit":"1ad0f2e7139c9214197531c17f229b9a75fe117b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69325/","number":"69325","commit":"4030a9966211ae6344037657b8f12a78146504c3"}]}
PIPELINE-->

kibanamachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69304/)
* Commit: 5718f77e25cdc6dc03065f2ee3edb281c8cd9ad9
* Flaky suites:
  * oss-pluginFunctional

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69304/testReport)

<details><summary>Plugin Functional Tests.test/plugin_functional/test_suites/core/route·ts.core route timeouts idle socket should timeout if servers response is too slow</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69304/testReport//junit/Plugin%20Functional%20Tests/test_plugin_functional_test_suites_core_route·ts/core_route_timeouts_idle_socket_should_timeout_if_servers_response_is_too_slow)


#### Standard Out
```
Failed Tests Reporter:
  - Test has not failed recently on tracked branches

[00:00:00]       │
[00:00:00]         └-: core
[00:00:00]           └-> "before all" hook
[00:00:00]           └-: route
[00:00:00]             └-> "before all" hook
[00:00:00]             └-: timeouts
[00:00:00]               └-> "before all" hook
[00:00:00]               └-: idle socket
[00:00:00]                 └-> "before all" hook
[00:00:00]                 └-> should timeout if payload sending has too long of an idle period
[00:00:00]                   └-> "before each" hook: global before each
[00:00:00]                   └- ✓ pass  (46ms) "core route timeouts idle socket should timeout if payload sending has too long of an idle period"
[00:00:00]                 └-> should not timeout if payload sending does not have too long of an idle period
[00:00:00]                   └-> "before each" hook: global before each
[00:00:00]                   └- ✓ pass  (222ms) "core route timeouts idle socket should not timeout if payload sending does not have too long of an idle period"
[00:00:00]                 └-> should timeout if servers response is too slow
[00:00:00]                   └-> "before each" hook: global before each
[00:00:00]                   │ info Taking screenshot "/dev/shm/workspace/parallel/4/kibana/test/functional/screenshots/failure/core route timeouts idle socket should timeout if servers response is too slow.png"
[00:00:01]                   │ info Current URL is: data:/,
[00:00:01]                   │ info Saving page source to: /dev/shm/workspace/parallel/4/kibana/test/plugin_functional/failure_debug/html/core route timeouts idle socket should timeout if servers response is too slow.html
[00:00:01]                   └- ✖ fail: core route timeouts idle socket should timeout if servers response is too slow
[00:00:01]                   │      Error: expected 'write ECONNRESET' to equal 'socket hang up'
[00:00:01]                   │       at Assertion.assert (packages/kbn-expect/expect.js:100:11)
[00:00:01]                   │       at Assertion.be.Assertion.equal (packages/kbn-expect/expect.js:227:8)
[00:00:01]                   │       at Assertion.be (packages/kbn-expect/expect.js:69:22)
[00:00:01]                   │       at result.then.err (test/plugin_functional/test_suites/core/route.ts:147:38)
[00:00:01]                   │       at process._tickCallback (internal/process/next_tick.js:68:7)
[00:00:01]                   │ 
[00:00:01]                   │ 

```

#### Stack Trace
```
Error: expected 'write ECONNRESET' to equal 'socket hang up'
    at Assertion.assert (packages/kbn-expect/expect.js:100:11)
    at Assertion.be.Assertion.equal (packages/kbn-expect/expect.js:227:8)
    at Assertion.be (packages/kbn-expect/expect.js:69:22)
    at result.then.err (test/plugin_functional/test_suites/core/route.ts:147:38)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```
</details>

---

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/5718f77e25cdc6dc03065f2ee3edb281c8cd9ad9) | diff | [baseline](https://github.com/elastic/kibana/commit/de4b502cb31855b2c9c062ae97f99228138af27e) |
| --- | --- | --- | --- |
| ingestPipelines | 401 | +11 | 390 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/5718f77e25cdc6dc03065f2ee3edb281c8cd9ad9) | diff | [baseline](https://github.com/elastic/kibana/commit/de4b502cb31855b2c9c062ae97f99228138af27e) |
| --- | --- | --- | --- |
| ingestPipelines | 673.9KB | +30.5KB | 643.4KB |

### History
* :broken_heart: [Build #69325](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69325/) failed 4030a9966211ae6344037657b8f12a78146504c3
* :broken_heart: [Build #69322](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69322/) failed 1ad0f2e7139c9214197531c17f229b9a75fe117b
* :green_heart: [Build #69108](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69108/) succeeded 6ecf83090df2283b93cafa142fb275a80a9d5f49
* :green_heart: [Build #68982](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68982/) succeeded 3e83391bf2ca98da7278eadf0af8967415e39518
* :green_heart: [Build #68703](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68703/) succeeded 4592909a99f821039f6a6a5dc342511d056e32ae

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68703/","number":"68703","commit":"4592909a99f821039f6a6a5dc342511d056e32ae"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68982/","number":"68982","commit":"3e83391bf2ca98da7278eadf0af8967415e39518"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69108/","number":"69108","commit":"6ecf83090df2283b93cafa142fb275a80a9d5f49"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69322/","number":"69322","commit":"1ad0f2e7139c9214197531c17f229b9a75fe117b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69325/","number":"69325","commit":"4030a9966211ae6344037657b8f12a78146504c3"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69304/","number":"69304","commit":"5718f77e25cdc6dc03065f2ee3edb281c8cd9ad9"}]}
PIPELINE-->

jloleysens commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69351/)
* Commit: aea2c0b2bae001923c421f11fb6bdc3690ef01ab

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/aea2c0b2bae001923c421f11fb6bdc3690ef01ab) | diff | [baseline](https://github.com/elastic/kibana/commit/77a434b2f79c01438234d39a7cc1f25ddecc9f76) |
| --- | --- | --- | --- |
| ingestPipelines | 401 | +11 | 390 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/aea2c0b2bae001923c421f11fb6bdc3690ef01ab) | diff | [baseline](https://github.com/elastic/kibana/commit/77a434b2f79c01438234d39a7cc1f25ddecc9f76) |
| --- | --- | --- | --- |
| ingestPipelines | 675.6KB | +32.2KB | 643.4KB |

### History
* :broken_heart: [Build #69325](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69325/) failed 4030a9966211ae6344037657b8f12a78146504c3
* :yellow_heart: [Build #69304](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69304/) was flaky 5718f77e25cdc6dc03065f2ee3edb281c8cd9ad9
* :broken_heart: [Build #69322](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69322/) failed 1ad0f2e7139c9214197531c17f229b9a75fe117b
* :green_heart: [Build #69108](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69108/) succeeded 6ecf83090df2283b93cafa142fb275a80a9d5f49
* :green_heart: [Build #68982](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68982/) succeeded 3e83391bf2ca98da7278eadf0af8967415e39518

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/68982/","number":"68982","commit":"3e83391bf2ca98da7278eadf0af8967415e39518"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69108/","number":"69108","commit":"6ecf83090df2283b93cafa142fb275a80a9d5f49"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69322/","number":"69322","commit":"1ad0f2e7139c9214197531c17f229b9a75fe117b"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69304/","number":"69304","commit":"5718f77e25cdc6dc03065f2ee3edb281c8cd9ad9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69325/","number":"69325","commit":"4030a9966211ae6344037657b8f12a78146504c3"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69351/","number":"69351","commit":"aea2c0b2bae001923c421f11fb6bdc3690ef01ab"}]}
PIPELINE-->

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/geoip.tsx:
```diff
@@ -0,0 +1,110 @@
+      'xpack.ingestPipelines.pipelineEditor.geoIPForm.firstOnlyFieldHelpText',
+      {
+        defaultMessage:
+          'If true, the first matching geo data is used, even if the field contains an array.',
```
Another fix to one of my earlier suggestions.
```suggestion
          'Use the first matching geo data, even if the field contains an array.',
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/enrich.tsx:
```diff
@@ -0,0 +1,264 @@
+    helpText: i18n.translate(
+      'xpack.ingestPipelines.pipelineEditor.enrichForm.overrideFieldHelpText',
+      {
+        defaultMessage: 'If true, the processor can overwrite pre-existing field values.',
```
This was my original suggestion, but `true` doesn't feel right here.
```suggestion
        defaultMessage: 'If enabled, the processor can overwrite pre-existing field values.',
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/grok.tsx:
```diff
@@ -0,0 +1,134 @@
+    helpText: i18n.translate(
+      'xpack.ingestPipelines.pipelineEditor.grokForm.traceMatchFieldHelpText',
+      {
+        defaultMessage: 'If true, include metadata about the matching expression in the document.',
```
A fix to one of my earlier suggestions.
```suggestion
        defaultMessage: 'Add metadata about the matching expression to the document',
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/fail.tsx:
```diff
@@ -0,0 +1,40 @@
+      defaultMessage: 'Message',
+    }),
+    helpText: i18n.translate('xpack.ingestPipelines.pipelineEditor.failForm.messageHelpText', {
+      defaultMessage: 'Error message returned by the processor.',
```
Period isn't needed.
```suggestion
      defaultMessage: 'Error message returned by the processor',
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/foreach.tsx:
```diff
@@ -0,0 +1,81 @@
+      defaultMessage: 'Processor',
+    }),
+    helpText: i18n.translate('xpack.ingestPipelines.pipelineEditor.foreachForm.processorHelpText', {
+      defaultMessage: 'Ingest processor to run on each array value.',
```
Period isn't needed.
```suggestion
      defaultMessage: 'Ingest processor to run on each array value',
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/foreach.tsx:
```diff
@@ -0,0 +1,81 @@
+      <FieldNameField
+        helpText={i18n.translate(
+          'xpack.ingestPipelines.pipelineEditor.failForm.fieldNameHelpText',
+          { defaultMessage: 'Field containing array values.' }
```
Period isn't needed.
```suggestion
          { defaultMessage: 'Field containing array values' }
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/geoip.tsx:
```diff
@@ -0,0 +1,110 @@
+      <FieldNameField
+        helpText={i18n.translate(
+          'xpack.ingestPipelines.pipelineEditor.geoIPForm.fieldNameHelpText',
+          { defaultMessage: 'Field containing an IP address for the geographical lookup.' }
```
Period isn't needed.
```suggestion
          { defaultMessage: 'Field containing an IP address for the geographical lookup' }
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/grok.tsx:
```diff
@@ -0,0 +1,134 @@
+      <FieldNameField
+        helpText={i18n.translate(
+          'xpack.ingestPipelines.pipelineEditor.grokForm.fieldNameHelpText',
+          { defaultMessage: 'The array field.' }
```
I don't think this is the right help text for this field.
```suggestion
          { defaultMessage: 'Field to search for matches' }
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/gsub.tsx:
```diff
@@ -6,89 +6,75 @@
+      <FieldNameField
+        helpText={i18n.translate(
+          'xpack.ingestPipelines.pipelineEditor.gsubForm.fieldNameHelpText',
+          { defaultMessage: 'Field to search for matches.' }
```
```suggestion
          { defaultMessage: 'Field to search for matches' }
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/gsub.tsx:
```diff
@@ -6,89 +6,75 @@
+    }),
+    deserializer: String,
+    helpText: i18n.translate('xpack.ingestPipelines.pipelineEditor.gsubForm.patternFieldHelpText', {
+      defaultMessage: 'Regular expression used to match text.',
```
Period isn't needed.
```suggestion
      defaultMessage: 'Regular expression used to match substrings in the field',
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/gsub.tsx:
```diff
@@ -6,89 +6,75 @@
+    }),
+    helpText: i18n.translate(
+      'xpack.ingestPipelines.pipelineEditor.gsubForm.replacementFieldHelpText',
+      { defaultMessage: 'Replacement text for matches.' }
```
```suggestion
      { defaultMessage: 'Replacement text for matches' }
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/html_strip.tsx:
```diff
@@ -0,0 +1,39 @@
+      <FieldNameField
+        helpText={i18n.translate(
+          'xpack.ingestPipelines.pipelineEditor.htmlStripForm.fieldNameHelpText',
+          { defaultMessage: 'Field to remove HTML tags from.' }
```
```suggestion
          { defaultMessage: 'Field to remove HTML tags from' }
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/inference.tsx:
```diff
@@ -0,0 +1,203 @@
+    helpText: i18n.translate(
+      'xpack.ingestPipelines.pipelineEditor.inferenceForm.modelIDFieldHelpText',
+      {
+        defaultMessage: 'ID of the model to infer against.',
```
```suggestion
        defaultMessage: 'ID of the model to infer against',
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/join.tsx:
```diff
@@ -0,0 +1,71 @@
+    helpText: i18n.translate(
+      'xpack.ingestPipelines.pipelineEditor.joinForm.separatorFieldHelpText',
+      {
+        defaultMessage: 'Separator character.',
```
```suggestion
        defaultMessage: 'Separator character',
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/join.tsx:
```diff
@@ -0,0 +1,71 @@
+      <FieldNameField
+        helpText={i18n.translate(
+          'xpack.ingestPipelines.pipelineEditor.joinForm.fieldNameHelpText',
+          { defaultMessage: 'Field containing array values to join.' }
```
Period isn't needed.
```suggestion
          { defaultMessage: 'Field containing array values to join' }
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/json.tsx:
```diff
@@ -0,0 +1,82 @@
+      <FieldNameField
+        helpText={i18n.translate(
+          'xpack.ingestPipelines.pipelineEditor.jsonForm.fieldNameHelpText',
+          { defaultMessage: 'Field to be parsed.' }
```
```suggestion
          { defaultMessage: 'Field to be parsed' }
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/json.tsx:
```diff
@@ -0,0 +1,82 @@
+      'xpack.ingestPipelines.pipelineEditor.jsonForm.addToRootFieldHelpText',
+      {
+        defaultMessage:
+          'If true, add the JSON object to the top level of the document. Cannot be combined with a target field.',
```
```suggestion
          'Add the JSON object to the top level of the document. Cannot be combined with a target field.',
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/json.tsx:
```diff
@@ -0,0 +1,82 @@
+        )}
+      />
+
+      <TargetField />
```
The help text for this field is wrong:

<img width="679" alt="Screen Shot 2020-08-20 at 9 32 34 AM" src="https://user-images.githubusercontent.com/40268737/90776448-19a80600-e2c8-11ea-8c8d-bc57edc27837.PNG">

Should be: `Field used to contain the JSON object.`

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/enrich.tsx:
```diff
@@ -0,0 +1,264 @@
+        helpText={i18n.translate(
+          'xpack.ingestPipelines.pipelineEditor.enrichForm.targetFieldHelpText',
+          {
+            defaultMessage: 'Field used to contain enrich data.',
```
Period isn't needed.
```suggestion
            defaultMessage: 'Field used to contain enrich data',
```

jrodewig commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/enrich.tsx:
```diff
@@ -0,0 +1,264 @@
+          helpText: (
+            <FormattedMessage
+              id="xpack.ingestPipelines.pipelineEditor.enrichForm.policyNameHelpText"
+              defaultMessage="Name of the {enrichPolicyLink}."
```
Period isn't needed.
```suggestion
              defaultMessage="Name of the {enrichPolicyLink}"
```

jloleysens commented:
@jrodewig Thanks for the review! I assumed the period was needed, but happy to follow your/EUIs recommendation as that is what I thought I was doing 😅 - thanks for explaining, TIL! I'll implement your copy recommendations and merge.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69415/)
* Commit: 56642c2017fa02a45d51d0e56ce3b115b06e4eed

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/56642c2017fa02a45d51d0e56ce3b115b06e4eed) | diff | [baseline](https://github.com/elastic/kibana/commit/77a434b2f79c01438234d39a7cc1f25ddecc9f76) |
| --- | --- | --- | --- |
| ingestPipelines | 401 | +11 | 390 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/56642c2017fa02a45d51d0e56ce3b115b06e4eed) | diff | [baseline](https://github.com/elastic/kibana/commit/77a434b2f79c01438234d39a7cc1f25ddecc9f76) |
| --- | --- | --- | --- |
| ingestPipelines | 676.1KB | +32.7KB | 643.4KB |

### History
* :green_heart: [Build #69351](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69351/) succeeded aea2c0b2bae001923c421f11fb6bdc3690ef01ab
* :broken_heart: [Build #69325](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69325/) failed 4030a9966211ae6344037657b8f12a78146504c3
* :yellow_heart: [Build #69304](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69304/) was flaky 5718f77e25cdc6dc03065f2ee3edb281c8cd9ad9
* :broken_heart: [Build #69322](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69322/) failed 1ad0f2e7139c9214197531c17f229b9a75fe117b
* :green_heart: [Build #69108](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69108/) succeeded 6ecf83090df2283b93cafa142fb275a80a9d5f49

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69108/","number":"69108","commit":"6ecf83090df2283b93cafa142fb275a80a9d5f49"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69322/","number":"69322","commit":"1ad0f2e7139c9214197531c17f229b9a75fe117b"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69304/","number":"69304","commit":"5718f77e25cdc6dc03065f2ee3edb281c8cd9ad9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69325/","number":"69325","commit":"4030a9966211ae6344037657b8f12a78146504c3"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69351/","number":"69351","commit":"aea2c0b2bae001923c421f11fb6bdc3690ef01ab"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69415/","number":"69415","commit":"56642c2017fa02a45d51d0e56ce3b115b06e4eed"}]}
PIPELINE-->

jloleysens commented:
@jrodewig No problem and thanks for getting back with this info!
